core/provider/service.go

```diff
@@ -69,61 +99,68 @@ func (s *Service) Create(p *domain.Provider) error {
 		}
 	}
 
-	if err := provider.CreateConfig(p.Config); err != nil {
+	if err := c.CreateConfig(p.Config); err != nil {
 		return err
 	}
 
-	if err := s.providerRepository.Create(p); err != nil {
+	if err := s.repository.Create(p); err != nil {
 		return err
 	}
 
+	if err := s.auditLogger.Log(ctx, AuditKeyCreate, p); err != nil {
+		s.logger.Error("failed to record audit log", "error", err)
+	}
+
 	go func() {
-		s.logger.Info(fmt.Sprintf("fetching resources for %s", p.URN))
-		resources, err := s.getResources(p)
+		s.logger.Info("fetching resources", "provider_urn", p.URN)
+		resources, err := s.getResources(ctx, p)
```

Comment:
any reason why we do this inside go routine?
since we are using the context propagated from the parent function, if this goroutine takes longer time and the context is cancelled earlier, these will get cancelled too.
therefore, it is not different whether using goroutine or not.

Reply:
we want to fire and forget `FetchResources` the moment a provider is created.

I see, wasn't aware of that. But using a new context instance also means losing the audit actor info from the original ctx right?

Reply:
yes but that is fine right? we don't need embedded data in ctx inside the go routine and the logic inside go routines does not need to be depend on the ctx, we can just create a new context with its own timeout and kill go routine if it reaches the timeout.

Reply:
I was thinking we need to keep the actor from the original ctx, but I think creating new context with `"system"` actor should be fine for this fetching resources post provider create 👍